src/ktls_stream.rs

```diff
@@ -300,6 +300,9 @@ where
             if let Err(e) = crate::ffi::send_close_notify(this.inner.as_raw_fd()) {
                 return Err(e).into();
             }
+            if *this.read_closed {
+                unsafe { libc::close(this.inner.as_raw_fd()) };
```

Comment:
unless I'm confused (it's 4AM), this is what `this.inner.poll_shutdown` ends up doing, so I don't think it's necessary